### PR TITLE
⚡️ Add block refresher

### PR DESCRIPTION
**Pull request by mj426382**

### CODE REVIEW

File: packages/core/src/constants/type/Config.ts [hunk 1]
```diff
@@ -71,6 +71,10 @@ export type FullConfig = {
    * Enables reconnecting to last used provider when user revisits the page.
    */
   autoConnect: boolean
+  /**
+   * Refresh standard calls each time the n-th block is mined.
+   */
+  refresh?: number | string
```
Comment (rzadp) [inline packages/core/src/constants/type/Config.ts:77]: Specify available strings exactly, not string.

**Review by rzadp — CHANGES_REQUESTED** [review]

File: packages/core/src/hooks/useCall.ts [hunk 2]
```diff
@@ -76,10 +77,15 @@ export function useCall<T extends TypedContract, MN extends ContractMethodNames<
...
   const chainId = useChainId({ queryParams })
-  const { isStatic } = queryParams
+  const { refresh } = useConfig()
 
   const rawCalls = useMemo(
-    () => calls.map((call) => (chainId !== undefined ? encodeCallData(call, chainId, isStatic) : undefined)),
+    () =>
+      calls.map((call) =>
+        chainId !== undefined
+          ? encodeCallData(call, chainId, { ...queryParams, refresh: queryParams.refresh ?? refresh })
```
Comment (rzadp) [inline packages/core/src/hooks/useCall.ts:86]: Can it be simplified into `{ refresh, ...queryParams }`?

Comment (mj426382) [reply packages/core/src/hooks/useCall.ts:86]: Yes, it is leftover after rename refreshPerBlocks to refresh

File: packages/core/src/constants/type/QueryParams.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ export interface QueryParams {
    */
   chainId?: ChainId
   isStatic?: boolean
+  refresh?: number | 'never' | 'everyBlock'
```
Comment (rzadp) [inline packages/core/src/constants/type/QueryParams.ts:15]:
```suggestion
  refresh?: Config['refresh']
```

We can tie the types into one place.

File: packages/core/src/providers/chainState/multiChainStates/provider.tsx [hunk 4]
```diff
@@ -82,6 +82,7 @@ export function MultiChainStateProvider({ children, multicallAddresses }: Props)
 
     const callsOnThisChain = uniqueCalls.filter((call) => call.chainId === chainId)
     if (callsOnThisChain.length === 0) {
+      dispatchCalls({ type: 'UPDATE_CALLS', calls, blockNumber, chainId })
```
Comment (rzadp) [inline packages/core/src/providers/chainState/multiChainStates/provider.tsx:85]: Comment why this is needed here?

File: packages/core/src/hooks/useCall.ts (hunk 2 above)
Comment (rzadp) [reply packages/core/src/hooks/useCall.ts:86]: Yes, then queryParams will override.

File: packages/core/src/providers/chainState/multiChainStates/provider.tsx [hunk 5]
```diff
@@ -81,9 +81,7 @@ export function MultiChainStateProvider({ children, multicallAddresses }: Props)
     }
 
     const callsOnThisChain = uniqueCalls.filter((call) => call.chainId === chainId)
-    if (callsOnThisChain.length === 0) {
```
Comment (rzadp) [inline packages/core/src/providers/chainState/multiChainStates/provider.tsx:84]: Why do we remove this?

Comment (mj426382) [reply packages/core/src/providers/chainState/multiChainStates/provider.tsx:84]: i put that on multicall function, to reduce duplicated code

**Pull request merged**